src/main/java/org/andersen/homework/util/collection/MyArrayList.java

```diff
@@ -0,0 +1,80 @@
+package org.andersen.homework.util.collection;
+
+import java.util.Arrays;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class MyArrayList<E> {
+
+  private static final Object[] DEFAULT_EMPTY_CAPACITY_ARRAY = {};
+
+  private Object[] array;
+
+  private int capacity;
+
+  private int elementsCount = 0;
+
+  public MyArrayList() {
+    this.array = DEFAULT_EMPTY_CAPACITY_ARRAY;
+    this.capacity = 0;
+  }
+
+  public MyArrayList(int initialCapacity) {
+    if (initialCapacity > 0) {
+      this.array = new Object[initialCapacity];
+    } else if (initialCapacity == 0) {
+      this.array = DEFAULT_EMPTY_CAPACITY_ARRAY;
+    } else {
+      throw new IllegalArgumentException("Illegal Capacity: " + initialCapacity);
+    }
+
+    this.capacity = initialCapacity;
+  }
+
+  @SuppressWarnings("unchecked")
+  public E get(int index) {
+    Objects.checkIndex(index, elementsCount);
+    if (index > elementsCount - 1) {
+      return null;
```

Comment:
In which case we return null? Could you provide test for this in Main class?(just sout the value after getting )

Reply:
You're right, this statement is redundant. `Objects.checkIndex(index, elementsCount)` already does all the work, seems like I just forget to remove it after having a better realization idea